packages/menu/src/Menu.ts

```diff
@@ -51,11 +70,61 @@ export class Menu extends SpectrumElement {
      * @private
      */
     public get childRole(): string {
-        return this.getAttribute('role') === 'menu' ? 'menuitem' : 'option';
+        const [selects, role] = this.resolvedSelectsAndRole;
+        if (role === 'menu' || role === 'group') {
+            switch (selects) {
+                case 'single':
+                    return 'menuitemradio';
+                case 'multiple':
+                    return 'menuitemcheckbox';
+                default:
+                    return 'menuitem';
+            }
+        } else {
+            return 'option';
+        }
+    }
+
+    private get resolvedSelects(): undefined | String {
+        return this.resolvedSelectsAndRole[0];
+    }
+
+    private get resolvedSelectsAndRole(): [String | undefined, String | null] {
+        if (this.selects === 'inherit') {
+            let parent = this.parentElement;
+            if (parent == null) {
+                const shadowRoot = this.getRootNode() as ShadowRoot;
+                parent = shadowRoot?.host as HTMLElement;
+            }
+            while (parent != null) {
+                if (parent.tagName === 'SP-MENU') {
```

Comment:
I thought `tagName` was always capitalized in HTML: https://developer.mozilla.org/en-US/docs/Web/API/Element/tagName

I might be able to re-work this so it's not needed if I get rid of the role event though.

Reply:
Yes, I'm just saying that you can use `localName` like `tagName` but lowercase...because I had no idea and I much prefer lowercase, but not because I expected a change on this.

Binding to a tag names does generally cause some trouble around extensions, but I'm not sure we need to pre-optimize for that.